### PR TITLE
NIFI-5830 RedisConnectionPoolService does not work with Standal…

### PR DESCRIPTION
**Pull request by javajefe**
…one Redis using non-localhost deployment

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [Y] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message? NIFI-5830

- [Y] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [Y] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [Y] Is your initial contribution a single, squashed commit?

### For code changes:
- [Y] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [N] Have you written or updated unit tests to verify your changes?
- [N/A] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [N/A] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [N/A] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [N/A] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [Y] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/main/java/org/apache/nifi/redis/service/RedisDistributedMapCacheClientService.java [hunk 1]
```diff
@@ -188,7 +189,7 @@ public void onDisabled() {
     public <K, V> void put(final K key, final V value, final Serializer<K> keySerializer, final Serializer<V> valueSerializer) throws IOException {
         withConnection(redisConnection -> {
             final Tuple<byte[],byte[]> kv = serialize(key, value, keySerializer, valueSerializer);
-            redisConnection.set(kv.getKey(), kv.getValue(), Expiration.seconds(ttl), null);
+            redisConnection.set(kv.getKey(), kv.getValue(), Expiration.seconds(ttl), RedisStringCommands.SetOption.UPSERT);
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/main/java/org/apache/nifi/redis/service/RedisDistributedMapCacheClientService.java]: Not sure that change is related to the reported issue, no?

Comment (javajefe) [reply nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/main/java/org/apache/nifi/redis/service/RedisDistributedMapCacheClientService.java]: You’re right, this solves NPE you ‘ll see disregarding to the rest of changes. But as the root cause of the issue is the same (migration to newer version of Spring Data Redis 2.x) and the result of RedisUtils fix is incomplete (you can’t use non-localhost standalone Redis) I’ve decided to fix it in the same  patch.

**Comment by javajefe** [comment]
Hi, I'm just checking should I fix/improve/explain anything more or not? Thx

**Comment by pvillard31** [comment]
Hey @javajefe - I just merged #3135, could you rebase this PR against master?

**Comment by javajefe** [comment]
Done!
However, I see Travis-CI build failed and I can't believe that's because of my PR (the problem is in org.apache.nifi.dbcp.DBCPServiceTest and it looks like an integration test rather than unit test, anyway there is no relation between DBCPService and RedisUtils)

**Comment by pvillard31** [comment]
Changes make sense to me, reproduced the issue with a Docker instance and confirmed this PR fixes the issue when using a standalone Redis instance. +1, merging to master, thanks @javajefe

**Pull request closed without merging**